### PR TITLE
Adds spatial average notebook

### PR DESCRIPTION
**Pull request by jasonb5**
Adds spatial average notebook. Attempts to compare different spatial averaging methods; Generic method, CDAT's cdutil/genutil averager, xCDAT's spatial and Xarrays weighted/mean.

### CODE REVIEW

**Comment by jasonb5** [comment]
@xCDAT/core-developers I've reconstructed my notebook. This is just an initial commit, it feels like it still needs to be flushed out some more. Let me know if you feel like something else needs to be tested or tested differently. Thanks!

**Review by tomvothecoder — APPROVED** [review]
Hey @jasonb5, I made some updates to the organization of the cells in `spatial_average_comparison.ipynb` and provided some more information (e.g., section headers, overall conclusion). I think these changes help the notebook flow better.

I also updated the env files and moves other notebooks around, so those changes can be ignored.

Once you review `spatial_average_comparison.ipynb` and agree with the changes, you can merge this PR.

**Comment by jasonb5** [comment]
@tomvothecoder The changes look good to me. I can't merge the changes, I don't have write access on the repo. Thanks!

**Pull request merged**
